packages/neon/neon_talk/lib/src/blocs/room.dart

```diff
@@ -321,6 +324,26 @@ class _TalkRoomBloc extends InteractiveBloc implements TalkRoomBloc {
     replyTo.add(null);
   }
 
+  @override
+  Future<void> deleteMessage(spreed.$ChatMessageInterface chatMessage) async {
+    await wrapAction(
+      () async {
+        final response = await account.client.spreed.chat.deleteMessage(
+          token: token,
+          messageId: chatMessage.id,
+        );
+
+        updateLastCommonRead(response.headers.xChatLastCommonRead);
+
+        final m = response.body.ocs.data;
+        updateLastKnownMessageId(m.id);
+
+        prependMessages([m]);
```

Comment:
This is bugged and the test data was unfortunately chosen in a way that hides it.

Reply:
Ok I was stupid and this was right, just the mock implementation of the API was wrong. I was already wondering why this API would return the deleted message and not the system message for it.